pytest_bdd/parsers.py

```diff
@@ -0,0 +1,113 @@
+"""Step parsers."""
+
+from __future__ import absolute_import
+import re as base_re
+
+import parse as base_parse
+from parse_type import cfparse as base_cfparse
+
+import pytest
+import six
+
+from .exceptions import InvalidStepParserError
+
+
+RE_TYPE = type(base_re.compile(''))
+
+
+class StepArgumentParser(object):
+
+    """Parser of the individual step arguments."""
+
+    def __init__(self, name):
+        self.name = name
+
+    def get_params(self, name):
+        """Get step parameters from the given step name."""
+        raise NotImplementedError()
+
+    def is_matching(self, name):
+        """Match given name with the step name."""
+        raise NotImplementedError()
+
+
+class re(StepArgumentParser):
+
+    """Regex step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile regex."""
+        super(re, self).__init__(name)
+        self.regex = base_re.compile(self.name, **kwargs)
+
+    def get_params(self, name):
+        """Get step parameters."""
+        return self.regex.match(name).groupdict()
+
+    def is_matching(self, name):
+        """Match given name with the step name."""
+        return bool(self.regex.match(name))
+
+
+class parse(StepArgumentParser):
+
+    """parse step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile parse expression."""
+        super(parse, self).__init__(name)
+        self.parser = base_parse.compile(self.name, **kwargs)
+
+    def get_params(self, name):
+        """Get step parameters."""
+        return self.parser.parse(name).named
+
+    def is_matching(self, name):
+        """Match given name with the step name."""
+        try:
+            return bool(self.parser.parse(name))
+        except ValueError:
+            return False
+
+
+class cfparse(parse):
+
+    """cfparse step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile parse expression."""
+        super(parse, self).__init__(name)
+        self.parser = base_cfparse.Parser(self.name, **kwargs)
+
+
+class string(StepArgumentParser):
```

Comment:
will rename to StepParser

On 7 January 2015 at 15:29, Oleg Pidsadnyi <notifications@github.com> wrote:

> In pytest_bdd/parsers.py
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96#discussion-diff-22588103>
> :
>
> > +            return bool(self.parser.parse(name))
> > +        except ValueError:
> > +            return False
> > +
> > +
> > +class cfparse(parse):
> > +
> > +    """cfparse step parser."""
> > +
> > +    def __init__(self, name, **kwargs):
> > +        """Compile parse expression."""
> > +        super(parse, self).__init__(name)
> > +        self.parser = base_cfparse.Parser(self.name, **kwargs)
> > +
> > +
> > +class string(StepArgumentParser):
>
> Still it is a bit strange. That ArgumentParser doesn't parse any
> arguments. This is more like exact marcher indeed. Maybe rebrand it to
> Pattern?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96/files#r22588103>.
>



-- 
Anatoly Bubenkov